core/iwasm/compilation/aot_compiler.c

```diff
@@ -2615,14 +2615,20 @@ verify_module(AOTCompContext *comp_ctx)
     char *msg = NULL;
     bool ret;
 
+    /*
+     * LLVMVerifyModule() return true if the module is broken.
+     */
     ret = LLVMVerifyModule(comp_ctx->module, LLVMPrintMessageAction, &msg);
-    if (!ret && msg) {
-        if (msg[0] != '\0') {
-            aot_set_last_error(msg);
+    if (ret) {
+        if (msg) {
+            if (msg[0] != '\0')
+                aot_set_last_error(msg);
+            else
+                aot_set_last_error("LLVM IR Module is broken");
             LLVMDisposeMessage(msg);
-            return false;
         }
-        LLVMDisposeMessage(msg);
+        aot_set_last_error("LLVM IR Module is broken");
```

Comment:
Had better `else aot_set_last_error("LLVM IR Module is broken");`, or it may overwrite L2625